src/Microsoft.OData.Client/ALinq/ExpressionWriter.cs

```diff
@@ -372,6 +373,22 @@ internal override Expression VisitMethodCall(MethodCallExpression m)
                         this.builder.Append(UriHelper.FORWARDSLASH).Append(UriHelper.DOLLARSIGN).Append(UriHelper.COUNT);
                         return m;
                     }
+                    else if (sequenceMethod == SequenceMethod.Contains)

```

Comment:
The if on this method is way too big, perhaps it can be extracted into smaller specialized methods.

Reply:
I'm not sure if that's a valid concern here? It's 8 lines of actual code and the rest are comments. If you look at the surrounding if-blocks, this if-block is actually one of the smaller blocks.

Reply:
I may have misunderstood. Did you mean that the entire method should be broken up? Because then I agree 100%. But still, I don't think it's up to me as a first-time contributor to refactor all of _VisitMethodCall_ so I'll mark this as resolved.

Reply:
Agree.

---
In reply to: [498118908](https://github.com/OData/odata.net/pull/1868#discussion_r498118908) [](ancestors = 498118908)

Reply:
Would be great to refactor if, but not something to address in this PR.

---
In reply to: [485146911](https://github.com/OData/odata.net/pull/1868#discussion_r485146911) [](ancestors = 485146911)